### PR TITLE
[ISSUE #1770] Solve the problem of duplicate URL-encoding

### LINKED ISSUE #1770: The encoded urlPath will be re-encoded in WebClientPlugin
## Describe the bug

webClient will make the encoded urlPath do URL-encode in WebClientPlugin, This will cause the request to be encoded again
## Environment

- ShenYu version(s): [e.g. 2.2.1]

## Debug logs

Suppose the url send to gateway is : /test/testWebClient?key=%E7%94%B5%E8%A7%86%E5%89%A7

After the WebClientPlugin processing, The url sent to the back-end service becomes：
/test/testWebClient?key=%25E7%2594%25B5%25E8%25A7%2586%25E5%2589%25A7 ，
This will cause backend service processing errors.
<details>
<summary>Click to expand</summary>

```
replace this line with your debug logs
```
</details>

### PR DESCRIPTION
**Pull request by SouthEastLJ**
// Describe your PR here; eg. Fixes #issueNo
fix #1770 

The webclient will encode the incoming string type parameters, which will cause an error, we need to use the URI object.

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor/).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java [hunk 1]
```diff
@@ -73,7 +76,13 @@ public WebClientPlugin(final WebClient webClient) {
...
         log.info("The request urlPath is {}, retryTimes is {}", urlPath, retryTimes);
         HttpMethod method = HttpMethod.valueOf(exchange.getRequest().getMethodValue());
-        WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(urlPath);
+        try {
+            uriObject = new URI(urlPath);
+        } catch (Exception e) {
+            log.error("build URI object error",e);
+            return WebFluxResultUtils.result(exchange, e);
+        }
+        WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(uriObject);
```
Comment (KevinClair) [inline shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java:85]: Please use method `java.net.URI#create`

Comment (SouthEastLJ) [reply shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java:79]: Has been modified according to your suggestion

Comment (KevinClair) [reply shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java:79]: WebClient.RequestBodySpec requestBodySpec = webClient.method(method).uri(URI.create(urlPath));

Comment (SouthEastLJ) [reply shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/WebClientPlugin.java]: ok, has been modified according to your suggestion

**Comment by KevinClair** [comment]
Thanks for your pull request.

**Review by KevinClair — APPROVED** [review]

**Pull request merged**